moesif-java-request/src/main/java/com/moesif/moesifjavarequest/MoesifSpringRequestInterceptor.java

```diff
@@ -0,0 +1,188 @@
+
+    public MoesifSpringRequestInterceptor(MoesifAPIClient moesifApi) {
+        this.moesifApi = moesifApi;
+        this.config = new MoesifRequestConfiguration();
+
+        this.moesifApi.getAPI().setShouldSyncAppConfig(true);
+    }
+
+    public MoesifSpringRequestInterceptor(String applicationId) {
+        this.moesifApi = new MoesifAPIClient(applicationId);
+        this.config = new MoesifRequestConfiguration();
+
+        this.moesifApi.getAPI().setShouldSyncAppConfig(true);
+    }
+
+    public MoesifSpringRequestInterceptor(MoesifAPIClient moesifApi, MoesifRequestConfiguration config) {
+        this(moesifApi);
+        this.config = config;
+    }
+
+    public MoesifSpringRequestInterceptor(String applicationId, MoesifRequestConfiguration config) {
+        this(applicationId);
+        this.config = config;
+    }
+
+    private EventRequestModel buildEventRequestModel(HttpRequest request, byte[] body) {
+        EventRequestBuilder eventRequestBuilder = new EventRequestBuilder();
+        Map<String, String> headers = new HashMap<String, String>(0);
+        eventRequestBuilder
+                .time(new Date())
+                .uri(request.getURI().toString())
+                .headers(request.getHeaders().toSingleValueMap())
+                .verb(request.getMethod().toString())
+                .ipAddress(IpAddress.getClientIp(
+                    request.getHeaders().toSingleValueMap(),
+                    null
+                ));
+
+        String content = new String(body);
+        if (content != null && !content.isEmpty()) {
+            BodyParser.BodyWrapper bodyWrapper = BodyParser.parseBody(headers, content);
+            eventRequestBuilder.body(bodyWrapper.body);
+            eventRequestBuilder.transferEncoding(bodyWrapper.transferEncoding);
+        }
+
+
+        return eventRequestBuilder.build();
+    }
+
+    private EventResponseModel buildEventResponseModel(MoesifClientHttpResponse response) throws IOException {
+        EventResponseBuilder eventResponseBuilder = new EventResponseBuilder();
+        Map<String, String> headers = response.getHeaders().toSingleValueMap();
+
+        int statusCode = 0;
+
+        try {
+            statusCode = response.getRawStatusCode();
+        } catch (Exception e) {
+            logger.warning(e.toString());
+        }
+
+        eventResponseBuilder
+                .time(new Date())
+                .status(statusCode)
+                .headers(headers);
+
+        String content = response.getBodyString();
+
+        if (content != null && !content.isEmpty()) {
+            BodyParser.BodyWrapper bodyWrapper = BodyParser.parseBody(headers, content);
+            eventResponseBuilder.body(bodyWrapper.body);
+            eventResponseBuilder.transferEncoding(bodyWrapper.transferEncoding);
+        }
+
+
+        return eventResponseBuilder.build();
+    }
+
+    private EventResponseModel buildEventResponseModel(Exception e) {
```

Comment:
@dgilling is this how we should handle requests that fail? I think they should be reported to moesif somehow. `500` probably isn't the right status code either.